src/source-af-xdp.c

```diff
@@ -829,10 +834,21 @@ static TmEcode ReceiveAFXDPLoop(ThreadVars *tv, void *data, void *slot)
  * \param tv pointer to ThreadVars
  * \param data pointer that gets cast into AFXDPPThreadVars for ptv
  */
+static SCMutex sync_deinit = SCMUTEX_INITIALIZER;
+
 static TmEcode ReceiveAFXDPThreadDeinit(ThreadVars *tv, void *data)
 {
     AFXDPThreadVars *ptv = (AFXDPThreadVars *)data;
 
+    /*
+    * If AF_XDP is enabled, the program must be detached before the AF_XDP sockets
+    * are closed to mitigate a bug that causes an IO_PAGEFAULT in linux kernel
+    * version 5.19, unknown as of now what other versions this affects.
+    */
+    SCMutexLock(&sync_deinit);
+    RunModeAFXDPRemoveProg();
```

Comment:
I think it would be best to now also move this whole function into this file. That way we keep the diff simpler and the logic local.